packages/nimble-components/src/menu-button/testing/menu-button.pageobject.ts

```diff
@@ -0,0 +1,87 @@
+import {
+    keyArrowDown,
+    keyArrowUp,
+    keyEnter,
+    keyEscape,
+    keySpace
+} from '@microsoft/fast-web-utilities';
+import type { MenuButton } from '..';
+import { createEventListener } from '../../utilities/tests/component';
+
+/**
+ * Page object for `nimble-menu-button` component to provide consistent ways
+ * of querying and interacting with the component during tests.
+ *
+ * This page object is intended to be used to interact with a `nimble-menu-button`,
+ * not the menu slotted into the button or the menu items.
+ */
+export class MenuButtonPageObject {
+    public constructor(protected readonly menuButtonElement: MenuButton) {}
+
+    public isOpen(): boolean {
+        return this.menuButtonElement.open;
+    }
+
+    public clickMenuButton(): void {
+        // Focus the menu button before calling click() because ordinarily a mouse click
+        // would bring focus to the button, but calling click() directly does not.
+        this.menuButtonElement.focus();
+        this.menuButtonElement.toggleButton!.control.click();
+    }
+
+    public async openMenu(): Promise<void> {
+        if (this.isOpen()) {
+            return;
+        }
+
+        const toggleListener = createEventListener(
+            this.menuButtonElement,
+            'toggle'
+        );
+        this.clickMenuButton();
+        await toggleListener.promise;
+    }
+
+    public pressEnterOnMenuButton(): void {
```

Comment:
Do we need the `OnMenuButton` suffix for these methods? I ask only because a semblance of a naming pattern has been established on the `SelectPageObject` for these kinds of actions, and they don't include a similar suffix (they're all `press<key>Key`).

Reply:
That's an interesting question. I added that suffix based on feedback from @m-akinc that it wasn't clear whether some of these functions were interacting with the button or menu/menu-items.

Reply:
I think our page objects should align with our composition. So I'd expect different page objects for the menu-button, menu, and potentially menu items. And with references to those instances I'd expect the disambiguation at the reference naming vs in the api, i.e. `menuButtonPageObject.open();`, `menuPageObject.pressArrowDown();`, `menuItemPageObject.click()`. I could maybe see convenience apis on menu for interacting with menu items but from menu button all the way to menu items seems too far?

Reply:
Maybe a comment on those functions would be enough to make it clear that, say, if you open the menu and call `pressArrowDown()`, it's being pressed on the toggle button and not the menu. Both of these are parts of the menu button component, so it wasn't totally clear to me. Alternatively, if the page object had a `menuPageObject` property, that might also make it clearer that _that_ was the API for interacting with the menu.

Reply:
Added short comment for each function and renamed the key press functions to be consistent with the select page object.